### DISCUSSION TITLE
how to complete all the dependencies?

### DISCUSSION THREAD

**Discussion by laptopmutia** [crivotz/nv-ide#19] [Q&A] [answered]
DEPENDENCIES   

- Neovim
- [Universal-ctags](https://github.com/universal-ctags/ctags) with JSON format
- [FZF](https://github.com/junegunn/fzf)
- [ripgrep](https://github.com/BurntSushi/ripgrep): ripgrep recursively searches directories for a regex pattern
- [Ripper-tags](https://github.com/tmm1/ripper-tags)
- [code-minimap](https://github.com/wfxr/code-minimap) for minimap
- [prettierd](https://github.com/fsouza/prettierd) for formatter.nvim
- Needed DAP, show lua/plugins/nvim-dap.lua
- Needed LSP, show lua/lsp/init.lua
- Terminal that supports ligatures for proper representation

how to do the last 3?

- Needed DAP ...
- Needed LSP ...
- Terminal that ...

I know the last one is to use font from nerdfonts but don't know how to the DAP and LSP anyone could help me?

I want to use it for ruby on rails project

**Comment by crivotz** [comment]
For LSP you can follow the [docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt) from nvim-lspconfig (for ruby you need to install solargraph gem, I usually had in my Gemfile under development or install automatically via asdf for every ruby version)

For DAP you can follow the [docs](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation) from nvim-dap (also here for ruby it's only a gem)

About terminal it's important that supports ligatures, it's not only a problem of nerdfont. I reccomend [wezterm](https://github.com/wez/wezterm), [st](https://st.suckless.org/) or [kitty](https://github.com/kovidgoyal/kitty)

**Comment by crivotz** [comment, marked as answer]
I added [mason.nvim](https://github.com/williamboman/mason.nvim) and [mason-lspconfig](https://github.com/williamboman/mason-lspconfig.nvim) which should make it much easier to install current and future dependencies
